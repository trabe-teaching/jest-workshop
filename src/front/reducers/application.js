import { combineReducers } from "redux";
import loading, * as fromLoading from "reducers/loading";
import error, * as fromError from "reducers/error";
import users, * as fromUsers from "reducers/users";

export default combineReducers({ loading, error, users });

export const isLoading = state => fromLoading.isLoading(state.loading);
export const hasError = state => fromError.hasError(state.error);
export const getLightSiders = state => fromUsers.getLightSiders(state.users);
export const getDarkSiders = state => fromUsers.getDarkSiders(state.users);
