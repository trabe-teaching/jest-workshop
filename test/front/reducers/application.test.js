import deepFreeze from "deepfreeze";
import application, { isLoading, hasError, getLightSiders, getDarkSiders } from "reducers/application";

jest.mock("redux", () => ({
  combineReducers: spec => spec,
}));

jest.mock("reducers/loading", () => ({
  __esModule: true,
  default: "loadingReducer",
  isLoading: x => `isLoading ${x}`,
}));

jest.mock("reducers/error", () => ({
  __esModule: true,
  default: "errorReducer",
  hasError: x => `hasError ${x}`,
}));

jest.mock("reducers/users", () => ({
  __esModule: true,
  default: "usersReducer",
  getLightSiders: x => `getLightSiders ${x}`,
  getDarkSiders: x => `getDarkSiders ${x}`,
}));

describe("front/reducers/application", () => {
  describe("reducer", () => {
    it("combines other reducers", () => {
      expect(application).toMatchSnapshot();
    });
  });

  const state = {
    error: "error",
    loading: "loading",
    users: "users",
  };

  deepFreeze(state);

  describe("#isLoading", () => {
    expect(isLoading(state)).toEqual("isLoading loading");
  });

  describe("#hasError", () => {
    expect(hasError(state)).toEqual("hasError error");
  });

  describe("#getLightSiders", () => {
    expect(getLightSiders(state)).toEqual("getLightSiders users");
  });

  describe("#getDarkSiders", () => {
    expect(getDarkSiders(state)).toEqual("getDarkSiders users");
  });
});
