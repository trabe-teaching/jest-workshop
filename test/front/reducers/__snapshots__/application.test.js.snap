// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`front/reducers/application reducer combines other reducers 1`] = `
Object {
  "error": "errorReducer",
  "loading": "loadingReducer",
  "users": "usersReducer",
}
`;
